admin/tool/urlpreview/index.php

```diff
@@ -38,27 +40,24 @@
     throw new moodle_exception('noguest');
 }
 
-$messageform = new \tool_urlpreview\form\message_form();
-
-
 echo $OUTPUT->header();
 if (isloggedin()) {
     echo '<h3>Welcome, ' . fullname($USER) . '</h3>';
 }
-
-$messageform->display();
-
-if ($data = $messageform->get_data()) {
-
-    // Fetch the URL and get its metadata.
-    $url = $data->url;
-    // Assuming the unfurl_store class is available.
+$templatedata = [
+    'action' => 'index.php',
+    'formClass' => 'form-class',
+    'urlLabel' => 'URL',
+    'submitValue' => 'Submit'
+];
+echo $OUTPUT->render_from_template('tool_urlpreview/form', $templatedata);
+
+if ($_SERVER['REQUEST_METHOD'] === 'GET' && isset($_GET['url'])) {
```

Comment:
this line is redundant, as optional_param does this for you. If the param is not present then $url will be set to ''